### PR TITLE
Add ampl package

### PR DESCRIPTION
**Pull request by robgics**
Trying this again from a separate branch.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ampl/package.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                 )
+
+    def url_for_version(self, version):
+        return "file://{0}/ampl.linux-intel64.{1}.tgz".format(os.getcwd(), version)
+
+    def setup_run_environment(self, env):
+        env.prepend_path("PATH", self.prefix)
+        env.prepend_path("PATH", join_path(self.prefix, 'amplide'))
+        env.prepend_path("CPATH", join_path(self.prefix, 'amplapi', 'include'))
+        env.prepend_path("LD_LIBRARY_PATH", join_path(self.prefix, 'amplapi', 'lib'))
```
Comment (haampie) [inline var/spack/repos/builtin/packages/ampl/package.py]: Why do you need to set `CPATH` and `LD_LIBRARY_PATH` in the run environment?

Comment (robgics) [reply var/spack/repos/builtin/packages/ampl/package.py]: I'm pretty sure it's a "legacy" hold on from how we used this software before.  Checking with the vendor now for some official information on what env settings are required.

**Review by haampie — APPROVED** [review]

**Review by haampie — DISMISSED** [review]

**Comment by haampie** [comment]
Can you fix

```
var/spack/repos/builtin/packages/ampl/package.py:7: [I100] Import statements are in the wrong order. 'import os' should be before 'from spack import *' and in a different group.
var/spack/repos/builtin/packages/ampl/package.py:7: [I201] Missing newline between import groups. 'import os' is identified as Stdlib and 'from spack import *' is identified as Application.
```
this check was recently added

**Review by haampie — APPROVED** [review]

**Comment by robgics** [comment]
I'm at a loss as how to move this thing forward.  I cannot make sense of the error or how it might be related to my package.

**Pull request merged**